promcda/configuration/configuration_validator.py

```diff
@@ -0,0 +1,320 @@
+import sys
+import logging
+
+import numpy as np
+import pandas as pd
+from typing import Tuple, Union
+
+from pandas.core import series
+
+from promcda.enums import PDFType
+from promcda.utils.utils_for_main import pop_indexed_elements, check_norm_sum_weights, randomly_sample_all_weights, \
+     randomly_sample_ix_weight
+
+log = logging.getLogger(__name__)
+logging.getLogger('PIL').setLevel(logging.WARNING)
+FORMATTER: str = '%(levelname)s: %(asctime)s - %(name)s - %(message)s'
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, format=FORMATTER)
+logger = logging.getLogger("ProMCDA")
+
+from typing import List
+
+
+def validate_configuration(
+    input_matrix: pd.DataFrame,
+    polarity: tuple,
+    weights: list,
+    marginal_distributions: tuple,
+    num_runs: int,
+    num_cores: int,
+    random_seed: int,
+    robustness_weights: bool,
+    robustness_single_weights: bool,
+    robustness_indicators: bool
+):
+    # Required parameters
+    if input_matrix is None:
+        raise ValueError("The parameter 'input_matrix' is required but was not provided.")
+    if not isinstance(input_matrix, pd.DataFrame):
+        raise TypeError(f"Expected 'input_matrix' to be a DataFrame, got {type(input_matrix).__name__}")
+    if polarity is None:
+        raise ValueError("The parameter 'polarity' is required but was not provided.")
+    if not isinstance(polarity, tuple):
+        raise TypeError(f"Expected 'polarity' to be a Tuple, got {type(polarity).__name__}")
+    if not all(isinstance(item, str) for item in polarity):
+        raise ValueError("All elements in 'polarity' must be strings.")
+
+    # Optional parameters
+    # Check the input matrix
+    if robustness_indicators is False:
+        num_indicators = input_matrix.shape[1]
+    else:
+        num_non_exact_and_non_poisson = (len(marginal_distributions) -
+                                         marginal_distributions.count(PDFType.EXACT.value) -
+                                         marginal_distributions.count(PDFType.POISSON.value))
+        num_indicators = (input_matrix.shape[1] - num_non_exact_and_non_poisson)
+
+    for param_name, param_value in {
+        'robustness_weights': robustness_weights,
```

Comment:
Hard coded strings, but I also expect this logic to be gone if you use proposed method to not asks booleans from user rather strings based on ranking-service's Open API definition